conans/test/functional/toolchains/cmake/test_cmake_toolchain_m1.py

```diff
@@ -64,8 +64,6 @@ def build(self):
     assert "CMAKE_SYSTEM_PROCESSOR: arm64" in client.out
     main_path = "./build/Release/main.app/main" if op_system == "iOS" \
         else "./build/Release/main"
-    client.run_command(main_path, assert_error=True)
-    assert "Bad CPU type in executable" in client.out
```

Comment:
A note on this test: 
This test is parameterised and produces two `armv8` executables, one for `iOS` and another one for `macOS`. When running on an Intel Mac, both are cases of cross-building. But when running on an Apple Silicon Mac, the `macOS` one is a straightforward case (no cross-building).

I do wonder if we are losing any code paths being tested by this small distinction.